### PR TITLE
solveset is now able to solve XFAIL test_issue_failing_pow

### PR DESCRIPTION
**Pull request by Shekharrajak**
Fixes https://github.com/sympy/sympy/issues/10214

### CODE REVIEW

**Comment by gxyd** [comment]
This is not the right way to go. First of all using `.subs` for substitution doesn't always do the necessary simplification for checking the equality with `0`. It would be necessary to use `simplify` with it.
Also this is quite inefficient way to do it. The solution of the problem should be some replacement for `real_root` that i introduced in the PR #9804 . You can read [this comment](https://github.com/sympy/sympy/pull/9804#discussion_r36629286).

**Comment by smichr** [comment]
> This is not the right way to go

agreed. This definitely needs a lower level fix, not patching at the return statement.

**Comment by Shekharrajak** [comment]
@gxyd  , thanks for your suggestions.When I tested locally it was passing testcases of solveset ,So I thought I should check with travis once.
Your comment is for  `x^{\frac{3}{5}}+1=0` ans is -1 .[wolframalpha](http://www.wolframalpha.com/input/?i=solve%28x**%283%2F5%29+%2B+1%29); showing wrong answer right ?

here also [wolframalpha](http://www.wolframalpha.com/input/?i=solve%28x**%283%2F2%29+%2B+4%29) giving wrong answer , it should be `Emapyset` ,we know this right ?

I previously solved this manually.I cross check using another online solve 1. [symbolab](https://www.symbolab.com/solver/step-by-step/x%5E%7B%5Cfrac%7B3%7D%7B5%7D%7D%2B1%3D0/?origin=enterkey) 
2.[symbolab](https://www.symbolab.com/solver/step-by-step/x%5E%7B%5Cfrac%7B3%7D%7B5%7D%7D%2B1%3D0/?origin=enterkey).

But I think I should try a different way to solve this issue.

**Comment by Shekharrajak** [comment]
It seems from  this comment](https://github.com/sympy/sympy/pull/9804#discussion_r36629286)
```
>>> expr = x**(S(3)/5) + 1
>>> expr.subs({x: -1})
1 + (-1)**(3/5)   # which is non-zero
```
`SymPy` returning wrong answer for this, it should give zero.
because `(-1)**(1/n)` if n is odd then answer is `-1` .
so here `(-1)**(1/5)` is `-1` and `(-1)**3`  = `-1` so it should return `0`.Correct me if I am wrong.

**Comment by gxyd** [comment]
As a general comment you mentioned:
> Wolfram giving wrong results

I say Wolfram is giving right answers everywhere, for every link you shared. So I say you are interpreting them in a wrong way.

**Comment by Shekharrajak** [comment]
2nd [wolframalph](http://www.wolframalpha.com/input/?i=solve%28x**%283%2F2%29+%2B+4%29)
it is showing : `-2*(-2)**(1/3)` and `2*((-1)**(2/3))*(2)**(1/3)`
if we put this answer into the equation `x**(3/2)` we get `4`. so `x**(3/2) +4` can't be zero. Isn't it ? 
I cross checked by scientific calculator.

**Comment by gxyd** [comment]
Like i mentioned in [this comment](https://github.com/sympy/sympy/pull/9804#discussion_r36629286) , finding the `power` of a number and then simplifying it is "not the same" as find the root of an equation.
@Shekharrajak please read what i mentioned a little more carefully first, if you have any questions in the comment i mentioned, then ask again.
I again stress that wolfram is doing everything right.

**Comment by Shekharrajak** [comment]
ping @gxyd @smichr

**Comment by smichr** [comment]
`remove` should not be necessary: just `set1 - set2` gives `set1` without anything in common with `set2`. Also, I think filtering solution should not be done where you have done it. It should be filtering results in the location where those results are being computed. I don't know if checksol is used elsewhere, but that makes me nervous.

**Comment by Shekharrajak** [comment]
@smichr , I have updated the code, please review now once.

**Comment by smichr** [comment]
Again, I don't like the use of `checksol` here; solveset is supposed to give the complete solution and if it can't, it should return a ConditionSet. `checksol` has no guarantee of proving that something is a solution. Is it used anywhere else in solveset?

**Comment by Shekharrajak** [comment]
Yeah in solveset [_solve_radical](https://github.com/sympy/sympy/blob/master/sympy/solvers/solveset.py#L508) and some places in `solve ` also.

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -637,6 +638,18 @@ def _solveset(f, symbol, domain, _check=False):
                       if isinstance(s, RootOf)
                       or domain_check(fx, symbol, s)])
 
+            if isinstance(result, FiniteSet) and \
+            not all(isinstance(x, (erf, erfinv, erfc, erfcinv)) for x in result.args):
+                for r in result:
+                    if not checksol(f, symbol, r):
+                        result = FiniteSet(*[x for x in result if x != r])
+                if result is EmptySet():
```
Comment (smichr) [inline sympy/solvers/solveset.py]: Since you are returning the result regardless of whether it is empty or not, a simple `return result` would suffice.

**Comment by smichr** [comment]
> Yeah in solveset _solve_radical and some places in solve also.

`solve` is a different issue since it doesn't guarantee that the solution is complete. Any use in solveset -- including the use in `_solve_radical` should return a ConditionSet if any solution cannot be verified with True or False as being a solution.

**Comment by Shekharrajak** [comment]
In  [`_solve_radical`](https://github.com/sympy/sympy/blob/master/sympy/solvers/solveset.py#L508), first it should check empty set then return `ConditionSet` .

**Comment by smichr** [comment]
I would do this:

```python
valid = []
for s in result:
    v = checksol(f, symbol, s)
    if v is None:
        return ConditionSet...
    if v:
        valid.append(v)
return FiniteSet(*valid)
```

**Pull request closed without merging**